### PR TITLE
Tweak the way headers are set

### PR DESCRIPTION
**Pull request by sergey-borovkov**
## GitHub issue or Phabricator ticket number?
https://github.com/Screenly/cli/issues/41
## Checklist before merging

- [x] If have added tests.
- [ ] Is this a new feature? If yes, please write one phrase about this update.
- [ ] I've attached relevant screenshots (if relevant).

### CODE REVIEW

File: src/commands.rs [hunk 1]
```diff
@@ -354,6 +354,36 @@ impl AssetCommand {
...
+        id: &str,
+        headers: Vec<(String, String)>,
+    ) -> anyhow::Result<(), CommandError> {
+        // if no headers provided update has nothing to do
+        if headers.is_empty() {
+            return Ok(());
+        }
+
+        let mut new_headers: HashMap<_, _> = headers.into_iter().collect();
+        let asset = self.get(id)?;
```
Comment (korvyashka) [inline src/commands.rs:368]: You can request headers only to reduce payload for get - /v4/assets?select=headers

File: src/commands.rs [hunk 2]
```diff
@@ -354,6 +354,36 @@ impl AssetCommand {
...
+        if let Some(assets) = asset.value.as_array() {
+            if assets.is_empty() {
+                return Err(CommandError::MissingField);
+            }
+            let headers = assets[0].get("headers").ok_or(CommandError::MissingField)?;
+            let old_headers = serde_json::from_value::<HashMap<String, String>>(headers.clone())?;
+            debug!("Old headers {:?}", &old_headers);
+            for (key, value) in old_headers {
+                new_headers.entry(key).or_insert(value);
```
Comment (korvyashka) [inline src/commands.rs:377]: Hm - do you need to serialize/deserialize old_headers to/from json to mutate? It could be HashMap

**Review by korvyashka — CHANGES_REQUESTED** [review]

File: src/commands.rs (hunk 1 above)
Comment (sergey-borovkov) [reply src/commands.rs:368]: But it's the only place only headers are needed, and a single asset I don't see the reason to complicate it

File: src/commands.rs (hunk 2 above)
Comment (sergey-borovkov) [reply src/commands.rs:377]: How else can I iterate over then ? I did not find a way if you know how please tell me

File: src/commands.rs (hunk 1 above)
Comment (korvyashka) [reply src/commands.rs:368]: got it

File: src/commands.rs (hunk 2 above)
Comment (korvyashka) [reply src/commands.rs:377]: HashMap supports iter method:
https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.iter

Comment (sergey-borovkov) [reply src/commands.rs:377]: What hashmap? It's serde Value

**Review by korvyashka — APPROVED** [review]

**Pull request merged**